### PR TITLE
Closes #3487 - Display PWA site controls notification

### LINKED ISSUE #3487: Display silent notification when PWA is open
Since no toolbar is shown when a Progressive Web App is open, we should surface some basic actions like copying the URL and refreshing. One way to do this is by displaying a notification with actions whenever a PWA is open.

If this is implemented, pull-to-refresh (#2289) is no longer a blocker as it will be possible to refresh a PWA. (We should still implement pull-to-refresh!)

### PR DESCRIPTION
**Pull request by NotWoods**
Builds on top of #3816, merge that first.

Adds a new silent site controls notification when a PWA is open. These controls show an action for copying the URL of the web app and an action for refreshing the page.

Collapsed:
Screenshot_20190722-151726

Expanded:
Screenshot_20190722-151731

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.

### CODE REVIEW

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt [hunk 1]
```diff
@@ -76,16 +77,14 @@ class WebAppLauncherActivity : AppCompatActivity(), CoroutineScope by MainScope(
...
-            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_MULTIPLE_TASK
+        val intent = Intent(ACTION_VIEW_PWA, startUrl).apply {
             `package` = packageName
         }
 
         try {
             startActivity(intent)
         } catch (e: ActivityNotFoundException) {
-            logger.error("Packages does not handle AbstractWebAppShellActivity intent. Can't launch web app.", e)
+            logger.error("Packages does not handle ACTION_VIEW_PWA intent. Can't launch as web app.", e)
```
Comment (Amejia481) [inline components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt]: I'm not sure if the intention was `$ACTION_VIEW_PWA`?

Comment (NotWoods) [reply components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt]: This is actually from #3816.
Browsers no longer need to subclass `AbstractWebAppShellActivity` for PWA support. The `WebAppIntentProcessor` checks for ACTION_VIEW_PWA instead.

**Comment by NotWoods** [comment]
@Amejia481 @jonalmeida #3816 has been merged so this PR is now ready 😄

**Comment by Amejia481** [comment]
👍

**Pull request merged**